File: CSharp/Applications/WinFormsLoggingDI/DataStores/MainControlsDataStore.cs
using LogViewer.Core;
using LogViewer.WinForms;

namespace WinFormsLoggingDI.DataStores;

public class MainControlsDataStore
{
    #region Constructors

    public MainControlsDataStore(LogViewerControl logViewer,
                                 ILogDataStore dataStore)
    {
        LogViewer = logViewer;
        DataStore = dataStore;
    }

    #endregion

    #region Properties

    public LogViewerControl LogViewer { get; }

    public ILogDataStore DataStore { get; set; }

    #endregion
}

File: VB/Applications/WinFormsLoggingVBDI/DataStores/MainControlsDataStore.vb
Imports LogViewerVB.Core
Imports LogViewerVB.WinForms

Public Class MainControlsDataStore

#Region "Constructors"

    Public Sub New(logViewer As LogViewerControl,
                   dataStore As ILogDataStore)

        Me.LogViewer = logViewer
        Me.DataStore = dataStore

    End Sub

#End Region

#Region "Properties"

    Public ReadOnly Property LogViewer As LogViewerControl

    Public Property DataStore As ILogDataStore

#End Region

End Class